Pod/Classes/LSTracer.m

```diff
@@ -137,32 +141,114 @@ - (LSSpan*)startSpan:(NSString*)operationName
                 tags:(NSDictionary*)tags
            startTime:(NSDate*)startTime {
     // No locking required
+    return [[LSSpan alloc] initWithTracer:self
+                            operationName:operationName
+                                   parent:parentSpan
+                                     tags:tags
+                                startTime:startTime];
+}
 
-    NSString* traceGUID;
-    if (parentSpan != nil) {
-        traceGUID = [parentSpan _getTag:@"join:trace_guid"];
-    }
-    if (traceGUID == nil) {
-        traceGUID = [LSUtil generateGUID];
-    }
-    NSMutableDictionary* newTags = [NSMutableDictionary dictionaryWithDictionary:tags];
-    [newTags setObject:traceGUID forKey:@"join:trace_guid"];
-
-    LSSpan* span = [[LSSpan alloc] initWithTracer:self
-                                    operationName:operationName
-                                           parent:parentSpan
-                                             tags:newTags
-                                        startTime:startTime];
-    return span;
+- (bool)inject:(LSSpan*)span format:(NSString*)format carrier:(id)carrier {
+    return [self inject:span format:format carrier:carrier error:nil];
 }
 
-- (void)inject:(LSSpan*)span format:(NSString*)format carrier:(id)carrier {
-    // TODO: implement this
+// These strings are used for TextMap inject and join.
+static NSString* kBasicTracerStatePrefix   = @"ot-tracer-";
+static NSString* kTraceIdKey               = @"ot-tracer-traceid";
+static NSString* kSpanIdKey                = @"ot-tracer-spanid";
+static NSString* kSampledKey               = @"ot-tracer-sampled";
+static NSString* kBasicTracerBaggagePrefix = @"ot-baggage-";
+
+- (bool)inject:(LSSpan*)span format:(NSString*)format carrier:(id)carrier error:(NSError* __autoreleasing *)outError {
+    if ([format isEqualToString:OTFormatTextMap]) {
+        NSMutableDictionary *dict = carrier;
+        [dict setObject:span.hexTraceId forKey:kTraceIdKey];
+        [dict setObject:span.hexSpanId forKey:kSpanIdKey];
+        [dict setObject:@"true" forKey:kSampledKey];
+        for (NSString* key in span.tags) {
+            [dict setObject:[span.tags objectForKey:key] forKey:[kBasicTracerBaggagePrefix stringByAppendingString:key]];
```

Comment:
@bensigelman , AFAIU, encoding the baggage items each as their own header won't work well in any sort of CORS situation. 

For one, arbitrary baggage is basically impossible as there is no wildcard support for `Access-Control-Allow-Headers` ([MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS#Access-Control-Allow-Headers), [Stackoverflow](http://stackoverflow.com/questions/13146892/cors-access-control-allow-headers-wildcard-being-ignored)). Every supported key will need to be listed out explicitly in the CORS headers list.  

This may not matter if the desired behavior is that baggage item *are* intended to be explicit.

However, this strikes me as overly restrictive if the span data is intended to survive a round-trip to an arbitrary, OpenTracing-supporting third-party service on a different domain.  It'd need to list out every possible baggage and trace key for every supported implementation (and every version of every implementation).